static/js/click_handlers.js

```diff
@@ -229,43 +231,50 @@ exports.initialize = function () {
         message_edit.start(row);
         e.stopPropagation();
         popovers.hide_all();
+        popovers.hide_text_options_popover();
     });
     $('body').on('click', '.always_visible_topic_edit,.on_hover_topic_edit', function (e) {
         var recipient_row = $(this).closest(".recipient_row");
         message_edit.start_topic_edit(recipient_row);
         e.stopPropagation();
         popovers.hide_all();
+        popovers.hide_text_options_popover();
     });
     $("body").on("click", ".topic_edit_save", function (e) {
         var recipient_row = $(this).closest(".recipient_row");
         message_edit.show_topic_edit_spinner(recipient_row);
         message_edit.save(recipient_row, true);
         e.stopPropagation();
         popovers.hide_all();
+        popovers.hide_text_options_popover();
     });
     $("body").on("click", ".topic_edit_cancel", function (e) {
         var recipient_row = $(this).closest(".recipient_row");
         current_msg_list.hide_edit_topic(recipient_row);
         e.stopPropagation();
         popovers.hide_all();
+        popovers.hide_text_options_popover();
     });
     $("body").on("click", ".message_edit_save", function (e) {
         var row = $(this).closest(".message_row");
         message_edit.save(row, false);
         e.stopPropagation();
         popovers.hide_all();
+        popovers.hide_text_options_popover();
```

Comment:
Why did you add this in a million places?  It should be just a change to `popovers.hide_all()`.

Reply:
That is what I tried the first time, but adding it there caused the popover to never show up.
The popover generation happens at mouseup, but there is a click handler to close all popovers when users click outside of them. 
```
// Dismiss popovers if the user has clicked outside them
        if ($('.popover-inner, #user-profile-modal, .emoji-info-popover, .app-main [class^="column-"].expanded').has(e.target).length === 0) {
            popovers.hide_all();
        }
```
Changing this to suit this specific popover caused other popovers to not close when they should have. So, I finally opted to add this separately.

Reply:
We definitely don't want to add this in a million places; it's a maintainability mess and bad for readability.  We should figure out what rule was causing it to never show up, and adjust that to handle these menus (e.g. one could pass an option to `popover.hide_all()` there).  

It seems to me like the situation we have is that there's some additional situation when this popover should be closed?  In that case, we should make `popovers.hide_all()` call `popvers.hide_text_options_popover`, and additionally, call `hide_text_options_popover` for the extra case.

Reply:
Fixed this by creating a new function called `popovers.hide_text_options_popover()` so that the click event doesn't close this popover while still being able to use `popovers.hide_all()` in all other places as usual.